### PR TITLE
Extended frigate

### PR DESCRIPTION
**Pull request by GrumpyMeow**
## Description of changes

This PR improves the Fritgate package:
* Added support for audio-recognition role
* Removes some settings which broke other installed packages ([issue](https://github.com/NixOS/nixpkgs/issues/320512)) 
* Added some missing tmpfiles config
* Added openFirewall option
* Resolved some replace-deprecation warnings

I'm a newbie contributor..

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by SigmaSquadron** [comment]
Welcome to Nixpkgs! Please separate your changes to the package in a commit named `frigate: <changes to the package>` and the changes to the module in a commit named `nixos/frigate: <changes to the module>`. Also, squash any fixup/formatting commits.

File: pkgs/applications/video/frigate/default.nix [hunk 1]
```diff
@@ -52,6 +52,14 @@ let
     url = "https://github.com/openvinotoolkit/open_model_zoo/raw/master/data/dataset_classes/coco_91cl_bkgr.txt";
     hash = "sha256-5Cj2vEiWR8Z9d2xBmVoLZuNRv4UOuxHSGZQWTJorXUQ=";
   };
+
+  # Audio model
+  # https://github.com/blakeblackshear/frigate/blob/v0.13.2/docker/main/Dockerfile#L104
+  audio_model = fetchurl {
+    url = "https://www.kaggle.com/api/v1/models/google/yamnet/tfLite/classification-tflite/1/download";
```
Comment (SigmaSquadron) [inline pkgs/applications/video/frigate/default.nix:59]: Where does this URL come from? The Dockerfile points to https://tfhub.dev/google/lite-model/yamnet/classification/tflite/1?lite-format=tflite.

File: pkgs/applications/video/frigate/default.nix [hunk 2]
```diff
@@ -85,31 +93,35 @@ python.pkgs.buildPythonApplication rec {
...
-      --replace "/cpu_model.tflite" "${tflite_cpu_model}" \
-      --replace "/edgetpu_model.tflite" "${tflite_edgetpu_model}"
+      --replace-warn "/cpu_model.tflite" "${tflite_cpu_model}" \
+      --replace-warn "/edgetpu_model.tflite" "${tflite_edgetpu_model}"
 
     substituteInPlace frigate/test/test_config.py \
-      --replace "(MODEL_CACHE_DIR" "('/build/model_cache'" \
-      --replace "/config/model_cache" "/build/model_cache"
+      --replace-warn "(MODEL_CACHE_DIR" "('/build/model_cache'" \
+      --replace-warn "/config/model_cache" "/build/model_cache"
```
Comment (SigmaSquadron) [inline pkgs/applications/video/frigate/default.nix:124]:
```diff
- --replace-warn
+ --replace-error
```
Warnings wouldn't prevent automatic package updates from breaking Frigate's functionality in unexpected ways.

**Review by mweinelt — CHANGES_REQUESTED** [review]
Thank you for your contribution, I think we might have to iterate a bit on the formalities, as was already pointed out.

**Comment by SigmaSquadron** [comment]
Hi, @GrumpyMeow. This PR was closed as there have been no updates for months, and its multiple reviews were not acted upon. It has been superceded by #357717; feel free to review that PR and ascertain that all important features from this PR are included in the new one. If #357717 has a reduced scope, consider opening new pull requests that add new features after that PR has been merged.

**Pull request closed without merging**